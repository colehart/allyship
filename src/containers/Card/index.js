import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { toggleSaved } from '../../actions';
import './Card.css';

export const Card = (props) => {
  const {
    source,
    author,
    title,
    description,
    url,
    urlToImage,
    published,
    isSaved,
    toggleSaved
  } = props

  return (
    <article className='Card'>
      <div className='c-title-group'>
        <h3 className='c-title'>{title}</h3>
        <div
          className={`c-save ${ isSaved ? 'is-saved' : '' }`}
          onClick={() => toggleSaved(title)}
        >
        </div>
      </div>
      <p className='c-byline'>{published} | {source} {`${author ? `| ${author}` : ''}`}</p>
      <img className='c-image' src={urlToImage} alt='provided illustration'/>
      <p className='c-description'>{description}</p>
      <a
        className='c-link'
        href={url}
        target='_blank'
        rel='noopener noreferrer'
      >
        Read article from source
      </a>
    </article>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  toggleSaved: (headline) => dispatch(toggleSaved(headline))
})

Card.propTypes = {
  toggleSaved: PropTypes.func
}

export default connect(null, mapDispatchToProps)(Card)