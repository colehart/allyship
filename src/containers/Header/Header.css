.Header {
  grid-row: 1 / 2;
  grid-column: 1 / 2;
  display: flex;
  justify-content: space-around;
  align-items: center;
  background-color: var(--denim);
  box-shadow: 1px 1px 1px 1px var(--denim);
}

.hidden {
  display: none;
}

.btn {
  width: 125px;
  height: 50px;
  background-color: var(--coral);
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  text-align: center;
  color: white;
}

.btn:hover,
.btn:focus,
.active {
  background-color: var(--turquoise);
}

.btn:active {
  background-color: #498D93;
}

.h-logo-group:active {
  opacity: 0.8;
}

.h-badge {
  font-weight: 900;
  margin-left: 10px;
}

.h-logo {
  height: 60px;
}

.h-logo-group,
.h-nav-group {
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-decoration: none;
  color: white;
}

.h-logo-group {
  height: 100px;
  width: 215px;
}

.h-nav-group {
  height: 200px;
  width: 300px;
}

.h-error {
  height: 25px;
  width: 100%;
  display: flex;
  justify-content: stretch;
  align-items: center;
  background-color: var(--coral);
}

@media screen and (max-width: 655px) {
  .h-catgory-group {
    width: 100%;
  }
}