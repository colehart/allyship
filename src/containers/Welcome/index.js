import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import { fetchStories } from '../../thunks/fetchStories';
import './Welcome.css';

export const Welcome = (props) => {
  const { isLoading } = props

  return (
    <main className='Welcome'>
      { isLoading ? <Loader /> : null }
      <section
        className={ isLoading ? 'hidden' : 'w-message'}
      >
        <h2>Welcome!</h2>
        <p>
          Allyship exists to help you find your way in the galaxy of knowledge and news any true ally must navigate.
        <br />
        <br />
          Research the latest news about a specific minority identity or issue.
        {/* , or choose your own query by entering a search keyword or phrase. */}
        </p>
      </section>
      <div
        className='w-category-group'
      >
        <Link
          to='/transgender'
          exact='true'
          className='btn'
          aria-label='Click here for the latest transgender news'
        >
          Transgender
        </Link>
        <Link
          to='/immigration'
          exact='true'
          className='btn'
          aria-label='Click here for the latest immigration news'
        >
          Immigration
        </Link>
        <Link
          to='/black-lives-matter'
          exact='true'
          aria-label='Click here for the latest black lives matter news'
          className='btn'
        >
          Black Lives Matter
        </Link>
      </div>
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  fetchStories: (url) => dispatch(fetchStories(url))
})

Welcome.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  fetchStories: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
