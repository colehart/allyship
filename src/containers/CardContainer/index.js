import React from 'react';
import Card from '../Card'
import './CardContainer.css';

export const CardContainer = () => {
  const cards = []

  return (
    <main className="CardContainer">
      { cards }
    </main>
  )
}
