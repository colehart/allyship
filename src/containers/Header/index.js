import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import shuttleIcon from '../../assets/images/space-shuttle.svg';
import './Header.css';

export const Header = () => {
  return (
    <div className='Header'>
      <Link
        to='/'
        className='h-logo-group'
      >
        <img
          className='h-logo'
          src={ shuttleIcon }
          alt='A space shuttle'
        />
        <h1>Allyship</h1>
      </Link>
      <nav className='h-nav-group'>
        <NavLink
          exact to='/'
          className="h-categories btn"
        >
          Categories
        </NavLink>
        <NavLink
          exact to='/saved'
          className="h-saved btn"
        >
          Saved
          <span className='h-badge'> 0</span>
        </NavLink>
      </nav>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  caughtError: state.caughtError,
  stories: state.stories
})

export default connect(mapStateToProps)(Header)