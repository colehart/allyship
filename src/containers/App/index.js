import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { Header } from '../Header';
import { Welcome } from '../Welcome';
import { CardContainer } from '../CardContainer';
import './App.css';

export const App = (props) => {
  const { isLoading } = props

  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Your ship has gone off course.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the orange buttons in this app to navigate to your desired destination.</p>
    </div>
  )

  return (
    <div className='App'>
      <Header />
      <Switch>
        <Route
          exact path='/'
          component={ Welcome }
        />
        <Route
          exact path='/saved'
          render={() => CardContainer }
        />
        <Route
          exact path='/transgender'
          render={() => CardContainer }
        />
        <Route
          exact path='/immigration'
          render={() => CardContainer }
        />
        <Route
          exact path='/black-lives-matter'
          render={() => CardContainer }
        />
        <Route
          component={noMatch}
        />
      </Switch>
      {/* <Route
        exact path='/search'
        render={() => CardContainer }
      /> */}
    </div>
  );
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
})

export default connect(mapStateToProps)(App);
