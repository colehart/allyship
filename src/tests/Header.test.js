import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Header, mapStateToProps } from '../containers/Header';
import { mockDefaultState, mockStories, mockSavedStories, mockFullState, mockError } from './testMocks'

describe('Header', () => {
  describe('Header Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Header stories={mockStories}/>)
    })

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('renders numSaved accurately', () => {
      wrapper = shallow(<Header stories={mockSavedStories} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('renders error message accurately', () => {
      wrapper = shallow(<Header caughtError={mockError} stories={mockStories} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('matchStateToProps', () => {
    it('should parse caughtError from state', () => {
      const expected1 = ''

      const mappedProps1 = mapStateToProps(mockDefaultState)
      expect(mappedProps1.caughtError).toEqual(expected1)

      const expected2 = 'Bad API key'
      const mappedProps2 = mapStateToProps(mockFullState)
      expect(mappedProps2.caughtError).toEqual(expected2)
    })

    it('should parse stories from state', () => {
      const expected1 = []

      const mappedProps1 = mapStateToProps(mockDefaultState)
      expect(mappedProps1.stories).toEqual(expected1)

      const expected2 = mockStories

      const mappedProps2 = mapStateToProps(mockFullState)
      expect(mappedProps2.stories).toEqual(expected2)
    })
  })
})