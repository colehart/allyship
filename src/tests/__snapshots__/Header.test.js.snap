// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<div
  className="Header"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="h-logo-group"
    to="/"
  >
    <img
      alt="A space shuttle"
      className="h-logo"
      src="space-shuttle.svg"
    />
    <h1>
      allyShip
    </h1>
  </NavLink>
  <nav
    className="h-nav-group"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="h-categories btn"
      to="/"
    >
      Categories
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="h-saved btn"
      to="/saved"
    >
      Saved
      <span
        className="h-badge"
      >
         0
      </span>
    </NavLink>
  </nav>
</div>
`;
